### PR TITLE
DISTMYSQL-185: Fix error management during relocation of replicas.

### PR DESCRIPTION
**Pull request by y-trudeau**
DISTMYSQL-185: Created a variable to record the errors in the GTID section of "relocateReplicasInternal" without changing the logic flow.  Also fixed a dependency in the docker system tests for libncurse5.

Passes Jenkins: https://ps80.cd.percona.com/job/mysql-orchestrator-pipeline/16/

Jira: https://jira.percona.com/browse/DISTMYSQL-185

### CODE REVIEW

File: docker/Dockerfile.system [hunk 1]
```diff
@@ -8,7 +8,7 @@ RUN echo "ci_env_repo: $ci_env_repo"
 RUN echo "ci_env_branch: $ci_env_branch"
 
 RUN apt-get update -q -y
-RUN apt-get install -y sudo haproxy python git jq rsync libaio1 libnuma1 default-mysql-client bsdmainutils less vim
+RUN apt-get install -y sudo haproxy python git jq rsync libaio1 libnuma1 default-mysql-client bsdmainutils less vim libncurses5
```
Comment (kamil-holubicki) [inline docker/Dockerfile.system:11]: Why this change was needed?

File: go/inst/instance_topology.go [hunk 2]
```diff
@@ -2847,7 +2860,7 @@ func relocateReplicasInternal(replicas [](*Instance), instance, other *Instance)
 	}
 
 	// Too complex
-	return nil, log.Errorf("Relocating %+v replicas of %+v below %+v turns to be too complex; please do it manually", len(replicas), instance.Key, other.Key), errs
+	return nil, log.Errorf("Relocating %+v replicas of %+v below %+v turns to be too complex; please do it manually. Additional Errors: %v", len(replicas), instance.Key, other.Key,gtidErrorsMsg), errs
```
Comment (kamil-holubicki) [inline go/inst/instance_topology.go:2863]: Formatting: missing space after coma

File: go/inst/instance_topology.go [hunk 3]
```diff
@@ -2814,6 +2814,7 @@ func relocateReplicasInternal(replicas [](*Instance), instance, other *Instance)
 		return RepointTo(replicas, &other.Key)
 	}
 	// GTID
+	gtidErrorsMsg := ""
```
Comment (kamil-holubicki) [inline go/inst/instance_topology.go:2817]: Maybe we could initialize it to "none". This way the final result will be
`Additional Errors: none`
instead of
`Additional Errors: `
in case of not additional errors.

As the alternative (just a proposal) we could format the final string without `Additional Errors:` part if there are not additional errors.

**Review by kamil-holubicki — COMMENTED** [review]
Looks fine. Just a few cosmetic changes.
Please squash commits.

File: go/inst/instance_topology.go (hunk 3 above)
Comment (y-trudeau) [reply go/inst/instance_topology.go:2817]: Indeed, good idea.  I'll revise

File: go/inst/instance_topology.go [hunk 4]
```diff
@@ -2860,7 +2860,11 @@ func relocateReplicasInternal(replicas [](*Instance), instance, other *Instance)
 	}
 
 	// Too complex
-	return nil, log.Errorf("Relocating %+v replicas of %+v below %+v turns to be too complex; please do it manually. Additional Errors: %v", len(replicas), instance.Key, other.Key,gtidErrorsMsg), errs
+	// if the len of gtidErrorsMsg is less than 21, no errors were added
+	if len(gtidErrorsMsg) < 21 {
```
Comment (kamil-holubicki) [inline go/inst/instance_topology.go:2864]: Why 21?
I suppose it is the check if something was added after 'Additional Errors: ' string.
If yes, my proposal is to change the implementation to
1. Intialize gtidErrorsMsg to empty string at the beginning (line 2817)
2. Do the job
3. check if len(gtidErrorMsg) > 0. If yes, do gtidErrorsMsg = "Additional Errors: " + gtidErrorsMsg

**Review by kamil-holubicki — CHANGES_REQUESTED** [review]

File: docker/Dockerfile.system (hunk 1 above)
Comment (y-trudeau) [reply docker/Dockerfile.system:11]: This is for the docker system tests to pass

File: go/inst/instance_topology.go (hunk 4 above)
Comment (y-trudeau) [reply go/inst/instance_topology.go]: Modified according to the above comments, passed Jenkins tests.

**Review by kamil-holubicki — APPROVED** [review]
